base64.v doesn't compile

**V version:** 0.1.3
**OS:** Linux Mint
**C compiler version:**  n/a

**What did you do?**
Run `make test` in `compiler/`

**What did you expect to see?**
Tests passing
 
**What did you see instead?**
`base64.v` didn't build, giving me this:
```
➜ make test
find .. -name '*_test.v' | xargs v {}
PREV==
NEXT=NAME
pass=2 fn=`base64__decode`

====================
It looks like you are building V. It is being frequently updated every day. Most likely there was a change that lead to this error. 
Try to run `git pull`, that will most likely fix it.
If `git pull` doesn't help, re-install V from source or download a precompiled binary from https://vlang.io
====================

panic: base64.v:28
unexpected token: `for`
Makefile:11: recipe for target 'test' failed
make: *** [test] Error 123
```
I'm not sure what's happening here, it seems like a weird compiler syntax bug.

I've pulled the compiler, yadda yadda yadda, and `v` itself builds fine.

--- a/compiler/scanner.v
+++ b/compiler/scanner.v
@@ -128,16 +128,17 @@ fn (s mut Scanner) skip_whitespace() {
 	// }
 }
 
-fn (s mut Scanner) get_until_start(pos int) string {
+fn (s mut Scanner) get_var_name(pos int) string {
 	mut pos_start = pos
-	for ; pos_start >= 0 && s.text[pos_start] != `\n`; pos_start-- {}
-	for ; pos_start < pos && (s.text[pos_start] == ` ` || s.text[pos_start] == `\n`); pos_start++ {}
+
+	for ; pos_start >= 0 && s.text[pos_start] != `\n` && s.text[pos_start] != `;`; pos_start-- {}
+	pos_start++
 	return s.text.substr(pos_start, pos)
 }
 
 // CAO stands for Compound Assignment Operators  (e.g '+=' )
 fn (s mut Scanner) cao_change(operator string) {
-	s.text = s.text.substr(0, s.pos - 2) + ' = ' + s.get_until_start(s.pos - 2) + ' ' + operator + s.text.substr(s.pos + 1, s.text.len)
+	s.text = s.text.substr(0, s.pos - 1) + ' = ' + s.get_var_name(s.pos - 1) + ' ' + operator + ' ' + s.text.substr(s.pos + 1, s.text.len)
 }
 
 fn (s mut Scanner) scan() ScanRes {

